src/main/groovy/org/spongepowered/gradle/ore/OreDeployTask.groovy

```diff
@@ -0,0 +1,151 @@
+import org.apache.http.client.HttpClient
+import org.apache.http.client.methods.HttpPost
+import org.apache.http.entity.mime.MultipartEntityBuilder
+import org.apache.http.entity.mime.content.FileBody
+import org.apache.http.entity.mime.content.StringBody
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.util.EntityUtils
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.InvalidUserDataException
+import org.gradle.api.artifacts.Configuration
+import org.gradle.api.artifacts.PublishArtifact
+import org.gradle.api.logging.Logger
+import org.gradle.api.tasks.TaskAction
+import org.gradle.plugins.signing.Signature
+
+import javax.net.ssl.SSLHandshakeException
+
+class OreDeployTask extends DefaultTask {
+
+    String instanceUrl = 'https://ore.spongepowered.org'
+    boolean recommended = true
+    String channel
+    String apiKey
+
+    Configuration deploy = project.configurations.archives
+
+    @TaskAction
+    void run() {
+        Logger logger = project.logger
+        def artifacts = deploy.allArtifacts
+        PublishArtifact plugin = artifacts.find { a -> (a.type == "jar") }
+        PublishArtifact sig = artifacts.find { a -> (a instanceof Signature) }
+        if (plugin == null) {
+            throw new InvalidUserDataException("Plugin file not found.")
+        }
+        if (sig == null) {
+            throw new InvalidUserDataException("Signature file not found.")
+        }
+
+        logger.quiet('Publishing ' + plugin.name + ' to ' + instanceUrl + '.')
+        logger.quiet('  Recommended: ' + recommended)
+        logger.quiet('  Channel: ' + channel)
+
+        if (instanceUrl.endsWith("/")) {
+            instanceUrl = instanceUrl.substring(0, instanceUrl.length() - 1)
+        }
+        URL projectUrl
+        try {
+            String pluginId = project.sponge.plugin.id
+            String str = instanceUrl + '/api/projects/' + pluginId + '/versions/' + project.version
+            projectUrl = new URL(str)
+            logger.debug('POST ' + str)
+        } catch (MalformedURLException e) {
+            throw new InvalidUserDataException("Invalid project URL", e)
+        }
+
+        if (apiKey == null) {
+            apiKey = project.property('oreDeploy.apiKey')
+        }
+        if (channel == null) {
+            throw new InvalidUserDataException("Missing channel name.")
+        }
+        HttpEntity requestEntity = MultipartEntityBuilder.create()
+            .addPart('apiKey', new StringBody(apiKey))
+            .addPart('channel', new StringBody(channel))
+            .addPart('recommended', new StringBody(recommended as String))
+            .addPart('pluginFile', new FileBody(plugin.file))
+            .addPart('pluginSig', new FileBody(sig.file))
+            .build()
+        logger.debug(requestEntity.toString())
+
+        HttpClient http = HttpClients.createDefault()
+        HttpPost post = new HttpPost(projectUrl.toURI())
```

Comment:
Yeah but I would rather catch the exception for a malformed URL to provide a more useful error.